src/main/java/com/Acrobot/ChestShop/Dependencies.java

```diff
@@ -111,7 +111,7 @@ private static boolean loadEconomy() {
         }
 
         ChestShop.getMetrics().addCustomChart(ChestShop.createStaticDrilldownStat("economyAdapter", plugin, Bukkit.getPluginManager().getPlugin(plugin).getDescription().getVersion()));
-        ChestShop.getMetrics().addCustomChart(ChestShop.createStaticDrilldownStat("economyPlugin", economy.getProviderInfo().getName(), economy.getProviderInfo().getVersion()));
+        economy.registerProviderChangeListener((EconomyAdapter.ProviderInfo provider) -> ChestShop.getMetrics().addCustomChart(ChestShop.createStaticDrilldownStat("economyPlugin", provider.getName(), provider.getVersion())));
```

Comment:
Custom charts are a set in the metrics class so this would lead to a new chart getting registered every time the provider changes. I'm unsure if this actually causes issues when submitting it but I feel like we should have a better approach to avoid that. It should be as simple as just checking if `economy#getProviderInfo` returns `null`when the `providingPlugin`tbh.

Reply:
So you're suggesting that we should update the chart whenever the provider changes. Now we have this problem.
Is it possible to set the metric once within the ServerLoadEvent?

Reply:
Ideally you would just return the info of the currently used provider whenever the metrics are submitted as the charts can take a Callable which queries the current value. This wasn't done for the economy plugin before though so there is no method for that, but you should be able to easily adjust the implementation of `ChestShop#createStaticDrilldownStat` to achieve that.

Only setting it on the ServerLoadEvent would make the metric wrong as it wouldn't use the current provider (although that is already what's happening right now but the best way to fix this issue imo)